examples/prometheus-metrics/src/main.rs

```diff
@@ -23,35 +23,62 @@ use std::{
 };
 use tracing_subscriber::{layer::SubscriberExt, util::SubscriberInitExt};
 
-#[tokio::main]
-async fn main() {
-    tracing_subscriber::registry()
-        .with(tracing_subscriber::EnvFilter::new(
-            std::env::var("RUST_LOG")
-                .unwrap_or_else(|_| "example_todos=debug,tower_http=debug".into()),
-        ))
-        .with(tracing_subscriber::fmt::layer())
-        .init();
-
+fn metrics_app() -> Router {
     let recorder_handle = setup_metrics_recorder();
+    Router::new()
+        .route("/metrics", get(move || ready(recorder_handle.render())))
+        .route_layer(middleware::from_fn(track_metrics))
```

Comment:
Probably don't need to track metrics on the metrics server :P

```suggestion
```

Reply:
You are absolutely right. Fixed in 2151bb6